src/network/core/os_abstraction.h

```diff
@@ -23,9 +23,32 @@
 #include <ws2tcpip.h>
 #include <windows.h>
 
-#define GET_LAST_ERROR() WSAGetLastError()
+/**
+ * Get the last error code from any of the OS's network functions.
+ * What it returns and when it is reset, is implementation defined.
+ * @return The last error code.
+ */
+#define NetworkGetLastError() WSAGetLastError()
 #undef EWOULDBLOCK
 #define EWOULDBLOCK WSAEWOULDBLOCK
+#undef ECONNRESET
+#define ECONNRESET WSAECONNRESET
+
+/**
+ * Return the string representation of the given error from the OS's network functions.
+ * @param error The error number (from \c NetworkGetLastError()).
+ * @return The error message, potentially an empty string but never \c nullptr.
+ */
+static inline const char *NetworkGetErrorString(int error)
+{
+	static char buffer[512];
+	if (FormatMessageA(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, NULL, error,
+			MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), buffer, sizeof(buffer), NULL) == 0) {
+		return "Unknown error";
```

Comment:
If the string is unknown, can it include the error number in the response instead?